skglm/solvers/prox_newton_solver.py

```diff
@@ -0,0 +1,219 @@
+
+        # 2) run prox newton on smaller subproblem
+        for epoch in range(max_epochs):
+            _max_pn_cd_epochs = 1 if epoch == 0 else max_pn_cd_epochs
+            pn_tol = 0 if epoch == 0 else tol
+            # TODO: support sparse matrices
+            _prox_newton_iter(
+                X, Xw, w, y, penalty, ws, min_pn_cd_epochs, _max_pn_cd_epochs,
+                max_backtrack, pn_tol)
+
+            w_acc, Xw_acc = accelerator.extrapolate(w, Xw)
+            p_obj = datafit.value(y, w, Xw) + penalty.value(w)
+            p_obj_acc = datafit.value(y, w_acc, Xw_acc) + penalty.value(w_acc)
+
+            if p_obj_acc < p_obj:
+                w[:] = w_acc
+                Xw[:] = Xw_acc
+                p_obj = p_obj_acc
+
+            if epoch % 5 == 0:  # check every 5 epochs, PN epochs are expensive
+                grad = construct_grad(X, y, w, Xw, datafit, ws)
+                opt_ws = penalty.subdiff_distance(w, grad, ws)
+                stop_crit_in = np.max(opt_ws)
+                if max(verbose - 1, 0):
+                    print(f"Epoch {epoch + 1}, objective {p_obj:.10f}, "
+                          f"stopping crit {stop_crit_in:.2e}")
+                tol_in = tol if ws_size == n_features else 0.3 * stop_crit
+                if stop_crit_in <= tol_in:
+                    if max(verbose - 1, 0):
+                        print("Early exit")
+                    break
+        obj_out.append(p_obj)
+    return w, np.array(obj_out), stop_crit
+
+
+@njit
+def _prox_newton_iter(
+        X, Xw, w, y, penalty, ws, min_cd_epochs, max_cd_epochs, max_backtrack, tol):
+    n_samples, ws_size = X.shape[0], len(ws)
+
+    hessian_diag = np.zeros(n_samples)  # hessian = X^T D X, with D = diag(f_i'')
+    grad_datafit = np.zeros(n_samples)  # gradient of F(Xw)
+
+    lc = np.zeros(ws_size)  # weighted Lipschitz constants
+    bias = np.zeros(ws_size)
+
+    for i in range(n_samples):
+        prob = 1. / (1. + np.exp(y[i] * Xw[i]))
+        # this supposes that the datafit is scaled by n_samples
+        hessian_diag[i] = prob * (1. - prob) / n_samples
+        grad_datafit[i] = -y[i] * prob / n_samples
+
+    for idx, j in enumerate(ws):
+        lc[idx] = ((X[:, j] ** 2) * hessian_diag).sum()
+        bias[idx] = X[:, j] @ grad_datafit
+
+    delta_w, X_delta_w = _newton_cd(
+        X, w, ws, hessian_diag, bias, lc, penalty, min_cd_epochs, max_cd_epochs, tol)
+    step_size = _backtrack_line_search(
+        w, Xw, delta_w, X_delta_w, ws, y, penalty, max_backtrack)
+
+    for idx, j in enumerate(ws):
+        w[j] += step_size * delta_w[idx]
+    Xw += step_size * X_delta_w
+
+
+@njit
+def _newton_cd(
+        X, w, ws, hessian_diag, bias, lc, penalty, min_epochs, max_epochs, tol):
+    delta_w, X_delta_w = np.zeros(len(ws)), np.zeros(X.shape[0])
+    for epoch in range(max_epochs):
+        sum_sq_hess_diff = 0
+        for idx, j in enumerate(ws):
+            stepsize = 1/lc[idx] if lc[idx] != 0 else 1000
```

Comment:
Do you have an intuition on the hardcoded value `1000` for the stepsize here?

Reply:
same as here https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/solvers/cd_solver.py#L380